### DISCUSSION TITLE
Telefonnumer input med bekræftelse via SMS

### DISCUSSION THREAD

**Discussion by kropsi** [detfaellesdesignsystem/dkfds-components#200] [Q&A]
Vi har taget jeres bibliotek i brug for at lave formular til offentlige institutioner. Vi forsøger at lave ting efter bogen, men i visse tilfælde er det ret svært at afkode hvordan I har tænkt tingene skal håndteres. For eksempel når man skal kun registrere telefonnumre der er valideret med SMS, så kræver det to trin, dvs. selve indtastning og bekræftelse med bekræftelseskode.

Hvordan vil det være bedst at håndtere det, så det passer med jeres guidelines? Vi har på nuværende tidspunkt følgende forslag.

![CleanShot 2022-10-31 at 15 00 57](https://user-images.githubusercontent.com/946662/199026012-f73b0cbe-8341-4eb8-bf82-cda98ba83fcc.png)

Jeg synes ikke at det er et smuk syn. Det er som om i denne situation vil det være godt med et separat flow (i card visning eller modal view) eller en specialtegnet komponente til formålet.

Hvad vil være mest passende her?

**Comment by detfaellesdesignsystem** [comment]
Hej kropsi

Mange tak for din henvendelse! På nuværende tidspunkt har vi ikke nogen retningslinjer for validering af telefonnumre. Vi noterer, at der er et behov, og lægger en opgave om det i vores backlog så vi på længere sigt kan udbygge vores pattern om angivelse af telefonnumre med dette ( https://designsystem.dk/eksempler/patterns/angivelse-af-telefonnummer/ ).

I mellemtiden kan vi kun opfordre til, at I brugertester jeres valgte løsning og meget gerne deler erfaringerne med os. Jeg ved, at teamet bag Digital Post ( https://post.borger.dk/ ) har lavet en løsning, hvor angivelsen af selve bekræftelseskoden sker i en modal. Modal-løsningen fungerede godt i brugertest, så det kan I evt. lade jer inspirere af.

Venlig hilsen
Mette
Det Fælles Designsystem

**Reply by kropsi** [reply]
Hej Mette,

Tak for dit svar og dit forslag. Jeg har undersøgt måden de har gjort det på

![CleanShot 2022-11-01 at 12 58 32@2x](https://user-images.githubusercontent.com/946662/199228431-d90dcfd6-88f9-4656-89b1-431023ce0f1c.png)

Vi vil prøve at designe det med jeres komponenter og dele vores erfaring med jer.
